django/river/settings/base.py

```diff
@@ -247,3 +251,32 @@
 # Prometheus
 
 EXPORTER_PORT = os.environ.get("EXPORTER_PORT", 8001)
+
+# Mozilla OpenID Connect
+# https://mozilla-django-oidc.readthedocs.io/en/stable/settings.html
+
+OIDC_STORE_ID_TOKEN = True
+OIDC_STORE_ACCESS_TOKEN = True
```

Comment:
Can you explain why storing the access token which is short lived ? And what about the refresh token ?

Reply:
I'm not sure if/why you think there is an issue.

The access token has an expiration time (`expires_in` field in a successful auth response). It can be an hour, it can be 12h. It's not a one shot token. Given the requests/response pattern of web frameworks (the framework will forget you in between requests) and the fact that we don't want to send the access token in the front (authz code flow), we need to store the access token in the session (which is store in DB).

Maybe you're asking me : why even bother storing the access token ?

1. asking a new access token for each requests will add latency and resources consumption. The provider will become more of a single point of failure.
2. IIRC access tokens have an expiration time bc it's otherwise hard to revoke them. Applications that still need an even longer access can use a refresh token. The refresh token mechanism forces a token bearer to check in with the provider, so that the provider can deny them access. Do you think our apps need to store the token longer than 12h ?
3. access token are not one-shot tokens, according to the spec. It goes against the spirit of the spec.
4. maybe thats why mozilla-django-oidc doesnt support storing refresh token.

Reply:
1. I don't understand this. It's not what's happening with mozilla oic if you don't store the access token. It won't try to get one at each request according to the code I've read. Or you can point me to where the access token is actually used after the session is created because I don't see it.
2. I agree with your point but I also think a refresh token can be useful in other ways (see my other comment)
3. Agreed
4. As they say in their readme, they keep it as minimal as possible.

Reply:
Regarding 1., you're also right. I was just saying that : not storing the access token implies that we'll need to make sure to fetch a new one each time a view requires access to a resource server. This has indeed nothing to do with mozilla-oidc. If many views need access to external resources, then we'd better take advantage of storing the access token and reuse it.